ethexe/contracts/src/Mirror.sol

```diff
@@ -172,14 +176,105 @@ contract Mirror is IMirror {
 
     /// @dev Value never sent since goes to mailbox.
     function _sendMailboxedMessage(Gear.Message calldata _message) private {
+        bytes calldata payload = _message.payload;
+
+        if (_message.destination == ETH_EVENT_ADDR && _message.value == 0 && payload.length > 0) {
+            uint256 topicsLength;
+            assembly ("memory-safe") {
+                topicsLength := shr(248, calldataload(payload.offset))
```

Comment:
an here: why 248 ?

Reply:
we need to load `uint8` to stack, but `calldataload` returns `uint256`, so this shift is needed to remove the extra bits.